src/enable-github-automerge-action.ts

```diff
@@ -0,0 +1,111 @@
+import { getInput, debug, error } from "@actions/core";
+import { GitHub } from "@actions/github/lib/utils";
+import { Context } from "@actions/github/lib/context"
+
+type GitHubClient = InstanceType<typeof GitHub>
+
+export default class EnableGithubAutomergeAction {
+
+    private client: GitHubClient
+  
+    private context: Context;
+  
+    constructor(client: GitHubClient, context: Context) {
+      this.client = client
+      this.context = context;
+    }
+  
+    async run() {
+    
+        // Find out where we are!
+        const { repo } = this.context;
+        if (!repo) {
+          throw new Error("Could not find repository!")
+        }
+    
+        // Make sure this is actually a pull-request!
+        // We need this to retrieve the pull-request node ID.
+        const { pull_request: pullRequest } = this.context.payload;
+        if (!pullRequest) {
+          throw new Error("Event payload missing `pull_request`, is this a pull-request?");
+        }
+        const pullRequestId = pullRequest.node_id
+  
+        // Step 1. Retrieve the merge method!
+        debug(`Retrieving merge-method...`)
+        const mergeMethod = await this.getMergeMethod(repo)
+        debug(`Successfully retrieved merge-method: ${mergeMethod}`)
+    
+        // Step 2. Enable auto-merge.
+        debug(`Enabling auto-merge for pull-request #${pullRequest.number}`);
+        await this.enableAutoMerge(pullRequestId, mergeMethod)
+        debug(
+          `Successfully enabled auto-merge for pull-request #${pullRequest.number}`
+        );
+  
+    }
+  
+    private async getMergeMethod(repo) {
+      
+      const preferredMergeMethod = getInput("merge-method", { required: false });
```

Comment:
This call to `getInput` isn't ideal - and probably should have happened in `main.ts`

Reply:
Perhaps it could be an optional argument to the constructor?

Reply:
By that I mean, make `preferredMergeMethod` an instance property which can be set by an optional argument in the constructor and then in the main.ts file you pass the value as `getInput("merge-method", { required: false })`

Reply:
It started it out that way, but it felt unnecessary...  Think I will add it back!  A few extra lines of code for a slight sounder sleep!

Reply:
Done!